csharp/src/Apache.Arrow/Ipc/ArrowStreamWriter.cs

```diff
@@ -248,6 +263,13 @@ private protected void WriteRecordBatchInternal(RecordBatch recordBatch)
                 HasWrittenSchema = true;
             }
 
+            if (!HasWrittenDictionaryBatch)
+            {
+                DictionaryCollector.Collect(recordBatch, ref _dictionaryMemo);
```

Comment:
Can you run the benchmark tests before and after this change to see what kind of impact this `DictionaryCollector` has when there are no dictionaries in the batch? Hopefully it isn't much, but I'd like to know.

Reply:
This is good news. Thanks for running the tests.